### PR TITLE
[QPPA-920] Refactor API Reference Nav (PR 1 of 4)

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
Refactor API Reference Navigation

[QPPA-920](https://jira.cms.gov/browse/QPPA-920)

Detail:
In preparation for adding a left nav bar to all pages, I create a [tech spec](https://docs.google.com/document/d/1CL-CXHcIJifPJYUkIQQaBN15nRuDl5vSC_f3CEJv0wc/edit#) which includes a refactoring of the navigation implementation so that it is simpler to extend to additional components. This is the first PR of 4 or 5 to implement the new navigation and styling on all pages.

Test plan:
* [x] adhoc testing via UI

aug-11-2017 14-00-07

### CODE REVIEW

File: src/components/api-reference/api-reference.jsx [hunk 1]
```diff
@@ -9,94 +9,82 @@ import Provider from './provider';
...
+  "provider-profile-stub": <Provider />
+};
+
+const SamplesNavBarItemsMap = {
+  "example-submission-JSON-&-XML": <ExampleDocs />
+};
+
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
```
Comment (kencheeto) [inline src/components/api-reference/api-reference.jsx]: nit: i'm like 3/10 for using `capitalize` from lodash

File: src/components/api-reference/api-reference.jsx [hunk 2]
```diff
@@ -9,94 +9,82 @@ import Provider from './provider';
...
+
+function keyToTitle(key) {
+  return key.split("-").map((string) => capitalize(string)).join(" ");
```
Comment (kencheeto) [inline src/components/api-reference/api-reference.jsx]: oo, so "provider-profile-stub" -> "Provider Profile Stub" ? clever, but i wonder if keeping data and logic separate would make it easier to change titles or add i18n later. would a static map of names to titles also work?

File: src/components/api-reference/api-reference.test.js [hunk 3]
```diff
@@ -13,12 +14,14 @@ it('renders submission schema by default', () => {
   expect(component.containsMatchingElement(<Submission />)).toEqual(true);
 });
 
-it('routes to benchmarks when the hash is #historical-benchmarks', () => {
-  const component = mount(<ApiReference hash='#historical-benchmarks' />);
+it('routes to benchmarks when the activeComponent is benchmarks', () => {
```
Comment (kencheeto) [inline src/components/api-reference/api-reference.test.js]: i left some suggestions on the tech spec, but I think it's important to preserve the hash-driven navigation if possible and to make path or hash routing possible if not.

File: src/components/api-reference/api-reference.jsx [hunk 4]
```diff
@@ -9,94 +9,82 @@ import Provider from './provider';
...
   render() {
+    var referenceNavListItems = [];
+    Object.keys(ReferenceNavBarItemsMap).forEach((itemName) => {
+      referenceNavListItems.push(
+        // the onClick property sends the NavBarListItem an onClick property which is the handleClick function
+        <NavBarListItem key={itemName} name={itemName} isActive={ this.state.activeComponent === itemName } onClick={this.handleClick} />
+      )
+    });
+    var samplesNavListItems = [];
+    Object.keys(SamplesNavBarItemsMap).forEach((itemName) => {
```
Comment (kencheeto) [inline src/components/api-reference/api-reference.jsx]: nit: could be terser:
```js
const samplesNavListItems = Object.keys(SamplesNavBarItemsMap).map(itemName => (
  <NavBarListItem
    key={itemName}
    name={itemName}
    ...>
));
```

File: src/components/api-reference/api-reference.jsx [hunk 5]
```diff
@@ -9,94 +9,82 @@ import Provider from './provider';
...
+      activeComponent: "submission"
     };
-
-    this.showSection = this.showSection.bind(this);
+    this.handleClick = this.handleClick.bind(this);
   }
 
-  showSection(event) {
-    this.setState({ hash: event.target.hash });
+  handleClick(e) {
```
Comment (kencheeto) [inline src/components/api-reference/api-reference.jsx]: nit: `handleClick` seems like a less descriptive name than `showSection`

**Comment by abarciauskas-bgse** [comment]
@kencheeto good call re stable links - I'm going to close and re-write some of the tech spec

**Pull request closed without merging**